microsoft365/onedrive/main.go

```diff
@@ -0,0 +1,95 @@
+	if len(os.Args) != 2 {
+		fmt.Println("Usage: Onedrive <command>")
+		os.Exit(1)
+	}
+
+	command := os.Args[1]
+
+	switch command {
+	case "listAllDrives":
+		if err := commands.ListAllDrives(context.Background()); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "getDrive":
+		if err := commands.GetDrive(context.Background(), os.Getenv("DRIVE_ID")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "listDriveItems":
+		if err := commands.ListDriveItems(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("FOLDER_ID")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "listSharedItems":
+		if err := commands.ListSharedItems(context.Background()); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "getDriveItem":
+		if err := commands.GetDriveItem(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("ITEM_ID")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "updateDriveItem":
+		if err := commands.UpdateDriveItem(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("ITEM_ID"), os.Getenv("NEW_FOLDER_ID"), os.Getenv("NEW_NAME")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "deleteDriveItem":
+		if err := commands.DeleteDriveItem(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("ITEM_ID")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "downloadDriveItem":
+		if err := commands.DownloadDriveItem(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("ITEM_ID"), os.Getenv("WORKSPACE_FILE_NAME")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "uploadDriveItem":
+		if err := commands.UploadDriveItem(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("FOLDER_ID"), os.Getenv("WORKSPACE_FILE_NAME")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "copyDriveItem":
+		if err := commands.CopyDriveItem(context.Background(), os.Getenv("SOURCE_DRIVE_ID"), os.Getenv("SOURCE_ITEM_ID"), 
+			os.Getenv("TARGET_DRIVE_ID"), os.Getenv("TARGET_FOLDER_ID"), os.Getenv("NEW_NAME")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "createFolder":
+		if err := commands.CreateFolder(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("FOLDER_ID"), os.Getenv("FOLDER_NAME")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "addPermission":
+		if err := commands.AddPermission(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("ITEM_ID"), os.Getenv("EMAILS"), os.Getenv("ROLE"), os.Getenv("MESSAGE"), os.Getenv("PASSWORD"), os.Getenv("EXPIRATION_DATE_TIME")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "deletePermission":
+		if err := commands.DeletePermission(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("ITEM_ID"), os.Getenv("PERMISSION_ID")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	case "listPermissions":
+		if err := commands.ListPermissions(context.Background(), os.Getenv("DRIVE_ID"), os.Getenv("ITEM_ID")); err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
```

Comment:
We could clean this up quite a lot by creating an `err` variable before the switch, and waiting until after it to check whether it is nil.